Lib/argparse.py

```diff
@@ -1697,6 +1697,28 @@ def add_mutually_exclusive_group(self, *args, **kwargs):
         return super().add_mutually_exclusive_group(*args, **kwargs)
 
 
+def _prog_name(prog=None):
+    if prog is not None:
+        return prog
+    arg0 = _sys.argv[0]
+    try:
+        modspec = _sys.modules['__main__'].__spec__
```

Comment:
```suggestion
        modspec = getattr(_sys.modules['__main__'], "__spec__", None)
```

`__spec__` not being set in `__main__` is a super weird edge case these days (usually involving some other piece of third party software doing the wrong thing), but it's easy to tolerate without throwing `AttributeError`, so we may as well do so.